src/Framework/AspNetCoreAnalyzers/src/Analyzers/RouteHandlers/RouteHandlerAnalyzer.cs

```diff
@@ -77,8 +77,11 @@ public override void Initialize(AnalysisContext context)
                     var foundMethodReferenceBody = false;
                     if (!methodReference.Method.DeclaringSyntaxReferences.IsEmpty)
                     {
-                        var syntaxReference = methodReference.Method.DeclaringSyntaxReferences[0];
-                        var methodOperation = invocation.SemanticModel.GetOperation(syntaxReference.GetSyntax(context.CancellationToken));
+                        var syntaxReference = methodReference.Method.DeclaringSyntaxReferences.SingleOrDefault();
+                        var syntaxNode = syntaxReference.GetSyntax(context.CancellationToken);
+                        var methodOperation = syntaxNode.SyntaxTree == invocation.SemanticModel.SyntaxTree
+                            ? invocation.SemanticModel.GetOperation(syntaxNode, context.CancellationToken)
+                            : null;
```

Comment:
It looks like the outcome of `methodOperation` doesn't really affect whether a diagnostic is produced or not, but affects the location where the diagnostic is produced.

In that case it's okay to return null if you don't care much about diagnostic location.

Reply:
Great!